File: pset6/readability/readability.py
def main():
    # Get input
    text = input("Text: ")
    GetGrade(text)


def GetGrade(text):
    # Characters that denote end of sentence
    sentenceEnd = [".", "!", "?"]
    letters = 0
    words = 0
    sentences = 0
    # Go throguh each character
    for t in text:
        # If it's a letter and has no spaces, increment the letter variable
        if t.isalpha() and t != " ":
            letters += 1
        # If it is a space, increment the word variable
        if t == " ":
            words += 1
        # If it is a sentence end increment the sentence variable
        if t in sentenceEnd:
            sentences += 1
    # Incrememnt for last word
    words += 1
    # Print grade given variables
    PrintGrade(letters, words, sentences)


def PrintGrade(letters, words, sentences):
    # Formula stuff
    l = (letters / words) * 100
    s = (sentences / words) * 100
    grade = round(0.0588 * l - 0.296 * s - 15.8)
    # If 16 or above, print 'Grade 16+'
    if grade >= 16:
        grade = "Grade 16+"
    # If less than one print 'Before Grade 1'
    elif grade < 1:
        grade = "Before Grade 1"
    # Else print current grade
    else:
        grade = "Grade: " + str(grade)
    print(grade)


main()

File: pset2/readability/readability.c
#include <stdio.h>
#include <string.h>
#include <math.h>
#include <ctype.h>
#include <cs50.h>

void GetTextInfo(string txt);
void CalculateGrade(int letters, int words, int sentences);

int main(void)
{
    // Get user input and pass it into a function
    string passageText = get_string("Text: ");
    GetTextInfo(passageText);
}

// Calculate the number of words, letters and sentences in the string
void GetTextInfo(string txt)
{
    int letters = 0, words = 0, sentences = 0;

    for (int i = 0, n = strlen(txt); i < n; i++)
    {
        // If it contains a character it's a letter
        if (isalpha(txt[i]))
        {
            letters++;
        }

        // If the char is a space it means it's a new word
        if (txt[i] == ' ')
        {
            words++;
        }

        // If it has a period, exclamation or question mark, it's the end of a sentence
        if (txt[i] == '.' || txt[i] == '!' || txt[i] == '?')
        {
            sentences++;
        }
    }
    // Last word
    words++;

    CalculateGrade(letters, words, sentences);
}

// Calculate the grade using Coleman-Liau index
void CalculateGrade(int letters, int words, int sentences)
{
    float l = (float) letters / words * 100;
    float s = (float) sentences / words * 100;

    float index = (0.0588 * l) - (0.296 * s) - 15.8;
    int grade = (int) round(index);

    if (grade > 16)
    {
        printf("Grade 16+\n");
    }

    else if (grade < 1)
    {
        printf("Before Grade 1\n");
    }

    else
    {
        printf("Grade %i\n", grade);
    }
}